pkg/controller/migplan/migplan_controller.go

```diff
@@ -389,43 +391,46 @@ func (r *ReconcileMigPlan) planSuspended(plan *migapi.MigPlan) error {
 
 func (r *ReconcileMigPlan) AddRunningLabelForPlan(plan *migapi.MigPlan) error {
 	migrations, err := plan.ListMigrations(r)
+	runningMigrations := 0
+	needsUpdate := false
 	if err != nil {
 		return liberr.Wrap(err)
 	}
 
-	if len(migrations) == 0 {
+	//Check if there are any associated migrations that are running
+	for _, m := range migrations {
+		if m.Status.HasCondition(migctl.Running) {
+			runningMigrations++
+		}
+	}
+
+	//No migrations associated with the migplan or there are no running migrations for the Migplan
+	if len(migrations) == 0 || runningMigrations == 0 {
```

Comment:
you only need `runningMigrations == 0` check right? why do you care about the number of migrations?

Reply:
I had added that just in case there is a stale running label associated with the migplan.

Reply:
if len(migrations) == 0 then the code will never to into this for loop https://github.com/konveyor/mig-controller/pull/681/files#diff-78c6d854c5e343bf8121a5e196e0d5b5R401 and runningMigrations will always be 0 right?

Reply:
anyway if you wanna keep it, I am fine, wont break anything IMO just an extra check